### ISSUE TITLE
RichTextLabel::get_content_height() gives garbage after setting bbcode_text

### ISSUE THREAD

**Issue by LunaticInAHat** [godotengine/godot#52634]
### Godot version

v3.4.beta.custom_build.42927e806

### System information

X11 (Manjaro)

### Issue description

After setting `bbcode_text` on a `RichTextLabel` (regardless of whether `bbcode_enabled` is set), the label's `get_content_height()` method will return an apparently-uninitialized value. This presents an issue in code like:
```
$RichTextLabel.bbcode_text = "Some [color=red]red[/color] text"
$RichTextLabel.rect_size.y = $RichTextLabel.get_content_height()
```
After the label has received `NOTIFICATION_DRAW`, the cached height accumulator gets updated to the correct value, however:
  - This doesn't help the case above, where the method is called before the label gets rendered
  - Setting `bbcode_text` again will re-uninitialize the height accumulator

### Steps to reproduce

I have attached a reproducer, however I don't think there's anything special about it; any usage of `RichTextLabel` looks like it will exhibit this issue.

In the attached reproducer, pressing play demonstrates that Panel is not being updated to have the expected size. Further, pressing any key will cause a further iteration of the logic, which results in the value reported by `get_content_height()` jumping around.
On my system, a typical run looks like:
```
Godot Engine v3.4.beta.custom_build.42927e806 - https://godotengine.org
OpenGL ES 3.0 Renderer: NVIDIA GeForce GTX 960/PCIe/SSE2
OpenGL ES Batching: ON
 
get_content_height(): 0
get_content_height(): 45
get_content_height(): 53
get_content_height(): 0
get_content_height(): 0
get_content_height(): 0
get_content_height(): 0
get_content_height(): 0
get_content_height(): 45
get_content_height(): 53
```

### Minimal reproduction project

[reproducer.zip](https://github.com/godotengine/godot/files/7155529/reproducer.zip)

**Comment by LunaticInAHat** [comment]
Naively, it seems like `set_bbcode()` should probably be calling `_validate_line_caches()` after updating its text, to ensure that all the cached values are legit. But I have by no means spent enough time inside the class to fully understand its design.

**Comment by qarmin** [comment]
When running project valgrind starts to shows hundreads of errors like:
```
==57336== Conditional jump or move depends on uninitialised value(s)
==57336==    at 0x377F780: Control::_size_changed() (control.cpp:1346)
==57336==    by 0x3780A1D: Control::set_margin(Margin, float) (control.cpp:1695)
==57336==    by 0x377FD94: Control::set_anchor_and_margin(Margin, float, float, bool) (control.cpp:1420)
==57336==    by 0x3893BE7: RichTextLabel::_update_scroll() (rich_text_label.cpp:923)
==57336==    by 0x389401C: RichTextLabel::_notification(int) (rich_text_label.cpp:996)
==57336==    by 0x38A7241: RichTextLabel::_notificationv(int, bool) (rich_text_label.h:38)
==57336==    by 0x46C48A4: Object::notification(int, bool) (object.cpp:927)
==57336==    by 0x3BFEF99: CanvasItem::_update_callback() (canvas_item.cpp:444)
==57336==    by 0x1863587: MethodBind0::call(Object*, Variant const**, int, Variant::CallError&) (method_bind.gen.inc:59)
==57336==    by 0x46C4765: Object::call(StringName const&, Variant const**, int, Variant::CallError&) (object.cpp:918)
==57336==    by 0x46BAD48: MessageQueue::_call_function(Object*, StringName const&, Variant const*, int, bool) (message_queue.cpp:241)
==57336==    by 0x46BB048: MessageQueue::flush() (message_queue.cpp:284)
==57336== 
==57336== Conditional jump or move depends on uninitialised value(s)
==57336==    at 0x38805C7: Range::set_value(double) (range.cpp:88)
==57336==    by 0x3880738: Range::set_max(double) (range.cpp:114)
==57336==    by 0x389777B: RichTextLabel::_validate_line_caches(RichTextLabel::ItemFrame*) (rich_text_label.cpp:1541)
==57336==    by 0x3893C41: RichTextLabel::_update_scroll() (rich_text_label.cpp:931)
==57336==    by 0x389401C: RichTextLabel::_notification(int) (rich_text_label.cpp:996)
==57336==    by 0x38A7241: RichTextLabel::_notificationv(int, bool) (rich_text_label.h:38)
==57336==    by 0x46C48A4: Object::notification(int, bool) (object.cpp:927)
==57336==    by 0x3BFEF99: CanvasItem::_update_callback() (canvas_item.cpp:444)
==57336==    by 0x1863587: MethodBind0::call(Object*, Variant const**, int, Variant::CallError&) (method_bind.gen.inc:59)
==57336==    by 0x46C4765: Object::call(StringName const&, Variant const**, int, Variant::CallError&) (object.cpp:918)
==57336==    by 0x46BAD48: MessageQueue::_call_function(Object*, StringName const&, Variant const*, int, bool) (message_queue.cpp:241)
==57336==    by 0x46BB048: MessageQueue::flush() (message_queue.cpp:284)

```

**Comment by qarmin** [comment]
Problem not happens with 4.0(valgrind also doesn't show anything)

**Comment by akien-mga** [comment]
Fixed by #56993.
